src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -208,7 +211,13 @@ public static boolean canContinueUsing(@NotNull ItemStack from, @NotNull ItemSta
         return false;
     }
 
-    public static boolean isCorrectToolForDrops(@NotNull BlockState state, @NotNull Player player) {
+    public static boolean isCorrectToolForDrops(@NotNull BlockGetter level, @NotNull BlockPos pos, @NotNull BlockState state, @NotNull Player player) {
+        if (level instanceof LevelAccessor accessor) { //BlockEvent expects a LevelAccessor
+            BlockEvent.ToolCheck event = EventHooks.doToolCheck(accessor, pos, state, player, player.getMainHandItem());
```

Comment:
```suggestion
            BlockEvent.ToolCheck event = NeoForge.EVENT_BUS.post(new BlockEntity.ToolCheck(accessor, pos, state, player, player.getMainHandItem()));
```
if you do this you can also remove the `EventHooks` method.

Reply:
I originally did it this way, then saw the pattern in EventHooks and adjusted to that. Will adjust back, one moment.

Reply:
Adjusted!